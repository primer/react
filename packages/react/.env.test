VITEST_SEGFAULT_RETRY=0
VITEST_SKIP_BROWSER=1