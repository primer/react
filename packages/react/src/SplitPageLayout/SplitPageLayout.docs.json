{
  "id": "split_page_layout",
  "name": "SplitPageLayout",
  "status": "alpha",
  "a11yReviewed": true,
  "stories": [],
  "importPath": "@primer/react",
  "props": [
    {
      "name": "sx",
      "type": "SystemStyleObject"
    }
  ],
  "subcomponents": [
    {
      "name": "SplitPageLayout.Header",
      "props": [
        {
          "name": "padding",
          "type": "| 'none' | 'condensed' | 'normal'",
          "defaultValue": "'normal'",
          "description": "The amount of padding inside the header."
        },
        {
          "name": "divider",
          "type": "| 'none' | 'line' | { narrow?: | 'none' | 'line' | 'filled' regular?: | 'none' | 'line' wide?: | 'none' | 'line' }",
          "defaultValue": "'line'",
          "description": ""
        },
        {
          "name": "hidden",
          "type": "| boolean | { narrow?: boolean regular?: boolean wide?: boolean }",
          "defaultValue": "false",
          "description": "Whether the header is hidden."
        },
        {
          "name": "sx",
          "type": "SystemStyleObject"
        }
      ]
    },
    {
      "name": "SplitPageLayout.Content",
      "props": [
        {
          "name": "width",
          "type": "| 'full' | 'medium' | 'large' | 'xlarge'",
          "defaultValue": "'large'",
          "description": "The maximum width of the content region."
        },
        {
          "name": "padding",
          "type": "| 'none' | 'condensed' | 'normal'",
          "defaultValue": "'normal'",
          "description": "The amount of padding inside the content."
        },
        {
          "name": "hidden",
          "type": "| boolean | { narrow?: boolean regular?: boolean wide?: boolean }",
          "defaultValue": "false",
          "description": "Whether the content is hidden."
        },
        {
          "name": "sx",
          "type": "SystemStyleObject"
        }
      ]
    },
    {
      "name": "SplitPageLayout.Pane",
      "props": [
        {
          "name": "width",
          "type": "| 'small' | 'medium' | 'large'",
          "defaultValue": "'medium'",
          "description": "The width of the pane."
        },
        {
          "name": "minWidth",
          "type": "number",
          "defaultValue": "256",
          "description": "The minimum width of the pane."
        },
        {
          "name": "resizable",
          "type": "boolean",
          "defaultValue": "false",
          "description": "When true, the pane may be resized by the user."
        },
        {
          "name": "widthStorageKey",
          "type": "string",
          "defaultValue": "'paneWidth'",
          "description": "Provide a key used by localStorage to persist the size of the pane on the client."
        },
        {
          "name": "sticky",
          "type": "boolean",
          "defaultValue": "true",
          "description": "Whether the pane should stick to the top of the screen while the content scrolls."
        },
        {
          "name": "offsetHeader",
          "type": "number | string",
          "defaultValue": "0",
          "description": "Use offsetHeader along with the sticky prop to push the sticky pane down to make room for a sticky header if necessary. Use the type `string` to specify the height with a unit i.e. 5rem; otherwise the type `number` will be taken as px."
        },
        {
          "name": "padding",
          "type": "| 'none' | 'condensed' | 'normal'",
          "defaultValue": "'normal'",
          "description": "The amount of padding inside the pane."
        },
        {
          "name": "divider",
          "type": "| 'none' | 'line' | { narrow?: | 'none' | 'line' | 'filled' regular?: | 'none' | 'line' wide?: | 'none' | 'line' }",
          "defaultValue": "'line'",
          "description": ""
        },
        {
          "name": "hidden",
          "type": "| boolean | { narrow?: boolean regular?: boolean wide?: boolean }",
          "defaultValue": "false",
          "description": "Whether the pane is hidden."
        },
        {
          "name": "sx",
          "type": "SystemStyleObject"
        }
      ]
    },
    {
      "name": "SplitPageLayout.Footer",
      "props": [
        {
          "name": "padding",
          "type": "| 'none' | 'condensed' | 'normal'",
          "defaultValue": "'normal'",
          "description": "The amount of padding inside the footer."
        },
        {
          "name": "divider",
          "type": "| 'none' | 'line' | { narrow?: | 'none' | 'line' | 'filled' regular?: | 'none' | 'line' wide?: | 'none' | 'line' }",
          "defaultValue": "'line'",
          "description": ""
        },
        {
          "name": "hidden",
          "type": "| boolean | { narrow?: boolean regular?: boolean wide?: boolean }",
          "defaultValue": "false",
          "description": "Whether the footer is hidden."
        },
        {
          "name": "sx",
          "type": "SystemStyleObject"
        }
      ]
    }
  ]
}
