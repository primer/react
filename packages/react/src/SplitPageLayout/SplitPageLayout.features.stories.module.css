.SectionHeading {
  /* stylelint-disable-next-line primer/typography */
  font-size: var(--base-size-24);
  font-weight: var(--base-text-weight-normal);
  color: var(--fgColor-danger);
  margin-bottom: var(--base-size-8);
}

.DeleteAccountContainer {
  border: var(--borderWidth-default) solid var(--borderColor-danger-emphasis);
  border-radius: var(--borderRadius-medium);
  padding: var(--base-size-16);
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: var(--stack-gap-normal);
}

.DeleteAccountTextContainer {
  display: grid;
  gap: var(--base-size-4);
}

.DeleteAccountTitle {
  font-size: var(--text-body-size-medium);
  font-weight: var(--base-text-weight-semibold);
  color: var(--fgColor-danger);
}

.DeleteAccountDescription {
  font-size: var(--text-body-size-medium);
  color: var(--fgColor-muted);
}
