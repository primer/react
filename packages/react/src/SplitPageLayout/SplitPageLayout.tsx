import React from 'react'
import type {SxProp} from '..'
import type {
  PageLayoutContentProps,
  PageLayoutFooterProps,
  PageLayoutHeaderProps,
  PageLayoutPaneProps,
} from '../PageLayout'
import {PageLayout} from '../PageLayout'

// ----------------------------------------------------------------------------
// SplitPageLayout

export type SplitPageLayoutProps = SxProp

export const Root: React.FC<React.PropsWithChildren<SplitPageLayoutProps>> = props => {
  return (
    <PageLayout
      containerWidth="full"
      padding="none"
      columnGap="none"
      rowGap="none"
      _slotsConfig={{
        header: Header,
        footer: Footer,
      }}
      {...props}
    />
  )
}

Root.displayName = 'SplitPageLayout'

// ----------------------------------------------------------------------------
// SplitPageLayout.Header

export type SplitPageLayoutHeaderProps = PageLayoutHeaderProps

export const Header: React.FC<React.PropsWithChildren<SplitPageLayoutHeaderProps>> = ({
  padding = 'normal',
  divider = 'line',
  ...props
}) => {
  // eslint-disable-next-line primer-react/direct-slot-children
  return <PageLayout.Header padding={padding} divider={divider} {...props} />
}

Header.displayName = 'SplitPageLayout.Header'

// ----------------------------------------------------------------------------
// SplitPageLayout.Content

export type SplitPageLayoutContentProps = PageLayoutContentProps

export const Content: React.FC<React.PropsWithChildren<SplitPageLayoutContentProps>> = ({
  width = 'large',
  padding = 'normal',
  ...props
}) => {
  return <PageLayout.Content width={width} padding={padding} {...props} />
}

Content.displayName = 'SplitPageLayout.Content'

// ----------------------------------------------------------------------------
// SplitPageLayout.Pane

export type SplitPageLayoutPaneProps = PageLayoutPaneProps

export const Pane: React.FC<React.PropsWithChildren<SplitPageLayoutPaneProps>> = ({
  position = 'start',
  sticky = true,
  padding = 'normal',
  divider = 'line',
  ...props
}) => {
  return (
    <PageLayout.Pane
      position={position}
      sticky={sticky}
      padding={padding}
      divider={divider}
      {...props}
    ></PageLayout.Pane>
  )
}
Pane.displayName = 'SplitPageLayout.Pane'

// ----------------------------------------------------------------------------
// SplitPageLayout.Footer

export type SplitPageLayoutFooterProps = PageLayoutFooterProps

export const Footer: React.FC<React.PropsWithChildren<SplitPageLayoutFooterProps>> = ({
  padding = 'normal',
  divider = 'line',
  ...props
}) => {
  // eslint-disable-next-line primer-react/direct-slot-children
  return <PageLayout.Footer padding={padding} divider={divider} {...props} />
}

Footer.displayName = 'SplitPageLayout.Footer'

// ----------------------------------------------------------------------------
// Export

export const SplitPageLayout = Object.assign(Root, {
  Header,
  Content,
  Pane,
  Footer,
})
