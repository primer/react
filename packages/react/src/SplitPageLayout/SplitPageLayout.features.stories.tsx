import React from 'react'
import type {StoryFn, Meta} from '@storybook/react'

import {Box, Button, Heading, Text} from '..'
import {NavList} from '../NavList'
import {SplitPageLayout} from '../SplitPageLayout'

export default {
  title: 'Components/SplitPageLayout/Features',
  component: SplitPageLayout,
} as Meta<typeof SplitPageLayout>

export const SettingsPage: StoryFn<typeof SplitPageLayout> = () => (
  <SplitPageLayout>
    <SplitPageLayout.Pane position="start">
      <NavList aria-label="Main navigation">
        <NavList.Item href="#">Profile</NavList.Item>
        <NavList.Item href="#" aria-current="page">
          Account
        </NavList.Item>
        <NavList.Item href="#">Emails</NavList.Item>
        <NavList.Item href="#">Notifications</NavList.Item>
      </NavList>
    </SplitPageLayout.Pane>
    <SplitPageLayout.Content>
      <Heading as="h2" sx={{fontSize: 4, fontWeight: 'normal', color: 'danger.fg', mb: 2}}>
        Danger zone
      </Heading>
      <Box
        sx={{
          border: '1px solid',
          borderColor: 'danger.emphasis',
          borderRadius: 2,
          p: 3,
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'space-between',
          gap: 3,
        }}
      >
        <Box sx={{display: 'grid', gap: 1}}>
          <Text sx={{fontSize: 1, fontWeight: 'bold', color: 'danger.fg'}}>Delete account</Text>
          <Text sx={{fontSize: 1, color: 'fg.muted'}}>
            Are you sure you don&apos;t want to just downgrade your account to a free account? We won&apos;t charge your
            credit card anymore.
          </Text>
        </Box>
        <Button variant="danger">Delete account</Button>
      </Box>
    </SplitPageLayout.Content>
  </SplitPageLayout>
)
