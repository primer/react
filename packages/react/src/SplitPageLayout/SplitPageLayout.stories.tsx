import React from 'react'
import type {Meta, StoryFn} from '@storybook/react'
import {Placeholder} from '../Placeholder'
import {SplitPageLayout} from '../SplitPageLayout'

export default {
  title: 'Components/SplitPageLayout',
  parameters: {
    layout: 'fullscreen',
    controls: {expanded: true},
  },
  argTypes: {
    // Debug controls
    'Render header?': {
      type: 'boolean',
      defaultValue: true,
      table: {category: 'Debug'},
    },
    'Render pane?': {
      type: 'boolean',
      defaultValue: true,
      table: {category: 'Debug'},
    },
    'Render footer?': {
      type: 'boolean',
      defaultValue: true,
      table: {category: 'Debug'},
    },
    'Header placeholder height': {
      type: 'number',
      defaultValue: 64,
      table: {category: 'Debug'},
    },
    'Pane placeholder height': {
      type: 'number',
      defaultValue: 200,
      table: {category: 'Debug'},
    },
    'Content placeholder height': {
      type: 'number',
      defaultValue: 400,
      table: {category: 'Debug'},
    },
    'Footer placeholder height': {
      type: 'number',
      defaultValue: 64,
      table: {category: 'Debug'},
    },

    // Header prop controls
    'Header.divider.regular': {
      type: {
        name: 'enum',
        value: ['none', 'line'],
      },
      defaultValue: 'line',
      control: {
        type: 'radio',
      },
      table: {
        category: 'Header props',
      },
    },
    'Header.divider.narrow': {
      type: {
        name: 'enum',
        value: ['none', 'line', 'filled'],
      },
      defaultValue: 'line',
      control: {
        type: 'radio',
      },
      table: {
        category: 'Header props',
      },
    },
    'Header.divider.wide': {
      type: {
        name: 'enum',
        value: ['none', 'line'],
      },
      defaultValue: 'line',
      control: {
        type: 'radio',
      },
      table: {
        category: 'Header props',
      },
    },
    'Header.padding': {
      type: {
        name: 'enum',
        value: ['none', 'condensed', 'normal'],
      },
      defaultValue: 'normal',
      control: {type: 'radio'},
      table: {category: 'Header props'},
    },
    'Header.hidden.regular': {
      type: 'boolean',
      defaultValue: false,
      table: {category: 'Header props'},
    },
    'Header.hidden.narrow': {
      type: 'boolean',
      defaultValue: false,
      table: {category: 'Header props'},
    },
    'Header.hidden.wide': {
      type: 'boolean',
      defaultValue: false,
      table: {category: 'Header props'},
    },

    // Content prop controls
    'Content.width': {
      type: {
        name: 'enum',
        value: ['full', 'medium', 'large', 'xlarge'],
      },
      defaultValue: 'xlarge',
      control: {type: 'radio'},
      table: {category: 'Content props'},
    },
    'Content.padding': {
      type: {
        name: 'enum',
        value: ['none', 'condensed', 'normal'],
      },
      defaultValue: 'normal',
      control: {type: 'radio'},
      table: {category: 'Content props'},
    },
    'Content.hidden.regular': {
      type: 'boolean',
      defaultValue: false,
      table: {category: 'Content props'},
    },
    'Content.hidden.narrow': {
      type: 'boolean',
      defaultValue: false,
      table: {category: 'Content props'},
    },
    'Content.hidden.wide': {
      type: 'boolean',
      defaultValue: false,
      table: {category: 'Content props'},
    },

    // Pane prop controls
    'Pane.position.regular': {
      type: {
        name: 'enum',
        value: ['start', 'end'],
      },
      defaultValue: 'start',
      control: {type: 'radio'},
      table: {category: 'Pane props'},
    },
    'Pane.id': {
      type: 'string',
      defaultValue: 'customId',
      table: {category: 'Pane props'},
    },
    'Pane.position.narrow': {
      type: {
        name: 'enum',
        value: ['start', 'end'],
      },
      defaultValue: 'start',
      control: {type: 'radio'},
      table: {category: 'Pane props'},
    },
    'Pane.position.wide': {
      type: {
        name: 'enum',
        value: ['start', 'end'],
      },
      defaultValue: 'start',
      control: {type: 'radio'},
      table: {category: 'Pane props'},
    },
    'Pane.width': {
      type: {
        name: 'enum',
        value: ['small', 'medium', 'large'],
      },
      defaultValue: 'medium',
      control: {type: 'radio'},
      table: {category: 'Pane props'},
    },
    'Pane.minWidth': {
      type: 'number',
      defaultValue: 256,
      table: {category: 'Pane props'},
    },
    'Pane.sticky': {
      type: 'boolean',
      defaultValue: true,
      table: {category: 'Pane props'},
    },
    'Pane.resizable': {
      type: 'boolean',
      defaultValue: false,
      table: {category: 'Pane props'},
    },
    'Pane.padding': {
      type: {
        name: 'enum',
        value: ['none', 'condensed', 'normal'],
      },
      defaultValue: 'normal',
      control: {type: 'radio'},
      table: {category: 'Pane props'},
    },
    'Pane.divider.regular': {
      type: {
        name: 'enum',
        value: ['none', 'line'],
      },
      defaultValue: 'line',
      control: {type: 'radio'},
      table: {category: 'Pane props'},
    },
    'Pane.divider.narrow': {
      type: {
        name: 'enum',
        value: ['none', 'line', 'filled'],
      },
      defaultValue: 'line',
      control: {type: 'radio'},
      table: {category: 'Pane props'},
    },
    'Pane.divider.wide': {
      type: {
        name: 'enum',
        value: ['none', 'line'],
      },
      defaultValue: 'line',
      control: {type: 'radio'},
      table: {category: 'Pane props'},
    },

    // Footer prop controls
    'Footer.divider.regular': {
      type: {
        name: 'enum',
        value: ['none', 'line'],
      },
      defaultValue: 'line',
      control: {
        type: 'radio',
      },
      table: {
        category: 'Footer props',
      },
    },
    'Footer.divider.narrow': {
      type: {
        name: 'enum',
        value: ['none', 'line', 'filled'],
      },
      defaultValue: 'line',
      control: {
        type: 'radio',
      },
      table: {
        category: 'Footer props',
      },
    },
    'Footer.divider.wide': {
      type: {
        name: 'enum',
        value: ['none', 'line'],
      },
      defaultValue: 'line',
      control: {
        type: 'radio',
      },
      table: {
        category: 'Footer props',
      },
    },
    'Footer.padding': {
      type: {
        name: 'enum',
        value: ['none', 'condensed', 'normal'],
      },
      defaultValue: 'normal',
      control: {type: 'radio'},
      table: {category: 'Footer props'},
    },
    'Footer.hidden.regular': {
      type: 'boolean',
      defaultValue: false,
      table: {category: 'Footer props'},
    },
    'Footer.hidden.narrow': {
      type: 'boolean',
      defaultValue: false,
      table: {category: 'Footer props'},
    },
    'Footer.hidden.wide': {
      type: 'boolean',
      defaultValue: false,
      table: {category: 'Footer props'},
    },
  },
} as Meta<typeof SplitPageLayout>

const Template: StoryFn = args => (
  <SplitPageLayout sx={args.sx}>
    {args['Render header?'] ? (
      <SplitPageLayout.Header
        padding={args['Header.padding']}
        divider={{
          narrow: args['Header.divider.narrow'],
          regular: args['Header.divider.regular'],
          wide: args['Header.divider.wide'],
        }}
        hidden={{
          narrow: args['Header.hidden.narrow'],
          regular: args['Header.hidden.regular'],
          wide: args['Header.hidden.wide'],
        }}
      >
        <Placeholder height={args['Header placeholder height']} label="Header" />
      </SplitPageLayout.Header>
    ) : null}
    <SplitPageLayout.Content
      width={args['Content.width']}
      padding={args['Content.padding']}
      hidden={{
        narrow: args['Content.hidden.narrow'],
        regular: args['Content.hidden.regular'],
        wide: args['Content.hidden.wide'],
      }}
    >
      <Placeholder height={args['Content placeholder height']} label="Content" />
    </SplitPageLayout.Content>
    {args['Render pane?'] ? (
      <SplitPageLayout.Pane
        resizable={args['Pane.resizable']}
        position={{
          narrow: args['Pane.position.narrow'],
          regular: args['Pane.position.regular'],
          wide: args['Pane.position.wide'],
        }}
        width={args['Pane.width']}
        minWidth={args['Pane.minWidth']}
        sticky={args['Pane.sticky']}
        padding={args['Pane.padding']}
        divider={{
          narrow: args['Pane.divider.narrow'],
          regular: args['Pane.divider.regular'],
          wide: args['Pane.divider.wide'],
        }}
        hidden={{
          narrow: args['Pane.hidden.narrow'],
          regular: args['Pane.hidden.regular'],
          wide: args['Pane.hidden.wide'],
        }}
        id={args['Pane.id']}
      >
        <Placeholder height={args['Pane placeholder height']} label="Pane" />
      </SplitPageLayout.Pane>
    ) : null}
    {args['Render footer?'] ? (
      <SplitPageLayout.Footer
        padding={args['Footer.padding']}
        divider={{
          narrow: args['Footer.divider.narrow'],
          regular: args['Footer.divider.regular'],
          wide: args['Footer.divider.wide'],
        }}
        hidden={{
          narrow: args['Footer.hidden.narrow'],
          regular: args['Footer.hidden.regular'],
          wide: args['Footer.hidden.wide'],
        }}
      >
        <Placeholder height={args['Footer placeholder height']} label="Footer" />
      </SplitPageLayout.Footer>
    ) : null}
  </SplitPageLayout>
)

export const Default = () => (
  <SplitPageLayout>
    <SplitPageLayout.Header>
      <Placeholder label="Header" height={100} />
    </SplitPageLayout.Header>
    <SplitPageLayout.Pane position="start">
      <Placeholder label="Pane" height={400} />
    </SplitPageLayout.Pane>
    <SplitPageLayout.Content>
      <Placeholder label="Content" height={600} />
    </SplitPageLayout.Content>
    <SplitPageLayout.Footer>
      <Placeholder label="Footer" height={100} />
    </SplitPageLayout.Footer>
  </SplitPageLayout>
)

export const Playground = Template.bind({})
