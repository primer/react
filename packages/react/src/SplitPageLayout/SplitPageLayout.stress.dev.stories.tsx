import React from 'react'

import type {Meta} from '@storybook/react'
import type {ComponentProps} from '../utils/types'
import {StressTest} from '../utils/StressTest'
import {SplitPageLayout} from './SplitPageLayout'
import Heading from '../Heading'
import {NavList} from '../NavList'

export default {
  title: 'StressTests/Components/SplitPageLayout',
  component: SplitPageLayout,
} as Meta<ComponentProps<typeof SplitPageLayout>>

const totalIterations = 20

const scrollPaneToTop = () => {
  const pane = document.getElementById('pane')
  pane?.scrollTo({
    top: 0,
    behavior: 'smooth',
  })
}
const scrollPaneToBottom = () => {
  const pane = document.getElementById('pane')
  pane?.scrollTo({
    top: pane.scrollHeight,
    behavior: 'smooth',
  })
}

export const ScrollPane = () => {
  function getItems() {
    return Array.from({length: 200}, (_, i) => (
      <NavList.Item key={i} href="#" aria-current={i === 0 ? 'page' : undefined}>
        {`Item ${i + 1}`}
      </NavList.Item>
    ))
  }

  return (
    <StressTest
      componentName="SplitPageLayout"
      title="Simple pane scroll"
      description="Scrolling pane up and down with a lot of items."
      totalIterations={totalIterations}
      renderIteration={() => (
        <SplitPageLayout>
          <SplitPageLayout.Pane id="pane" position="start" aria-label="Navigation Pane">
            <Heading as="h2">Pane</Heading>
            <NavList aria-label="Pane navlist">{getItems()}</NavList>
          </SplitPageLayout.Pane>
          <SplitPageLayout.Content>
            <Heading as="h2">Content</Heading>
            <NavList aria-label="Content navlist">{getItems()}</NavList>
          </SplitPageLayout.Content>
        </SplitPageLayout>
      )}
      onIteration={count => {
        count % 2 === 0 ? scrollPaneToBottom() : scrollPaneToTop()
      }}
      ms={1000}
    />
  )
}
