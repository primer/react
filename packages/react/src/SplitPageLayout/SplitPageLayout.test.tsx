import React from 'react'
import {render} from '@testing-library/react'
import MatchMediaMock from 'jest-matchmedia-mock'
import 'react-intersection-observer/test-utils'
import {ThemeProvider} from '..'
import {SplitPageLayout} from '../SplitPageLayout/SplitPageLayout'

let matchMedia: MatchMediaMock

describe('SplitPageLayout', () => {
  beforeAll(() => {
    matchMedia = new MatchMediaMock()
  })

  afterEach(() => {
    matchMedia.clear()
  })

  it('renders default layout', () => {
    const {container} = render(
      <ThemeProvider>
        <SplitPageLayout>
          <SplitPageLayout.Header>Header</SplitPageLayout.Header>
          <SplitPageLayout.Content>Content</SplitPageLayout.Content>
          <SplitPageLayout.Pane>Pane</SplitPageLayout.Pane>
          <SplitPageLayout.Footer>Footer</SplitPageLayout.Footer>
        </SplitPageLayout>
      </ThemeProvider>,
    )

    expect(container).toMatchSnapshot()
  })

  it('renders Pane with a custom ID', () => {
    const {getByText} = render(
      <ThemeProvider>
        <SplitPageLayout>
          <SplitPageLayout.Pane id="customId">Pane Content</SplitPageLayout.Pane>
        </SplitPageLayout>
      </ThemeProvider>,
    )

    const pane = getByText('Pane Content')

    expect(pane.getAttribute('id')).toBe('customId')
  })

  it('applies custom className', () => {
    const {container} = render(
      <ThemeProvider>
        <SplitPageLayout className="custom-class">
          <SplitPageLayout.Header>Header</SplitPageLayout.Header>
          <SplitPageLayout.Content>Content</SplitPageLayout.Content>
          <SplitPageLayout.Pane>Pane</SplitPageLayout.Pane>
          <SplitPageLayout.Footer>Footer</SplitPageLayout.Footer>
        </SplitPageLayout>
      </ThemeProvider>,
    )

    expect(container.firstChild).toHaveClass('custom-class')
  })
})
