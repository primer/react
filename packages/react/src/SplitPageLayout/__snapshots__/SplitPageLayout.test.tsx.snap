// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitPageLayout renders default layout 1`] = `
.c0 {
  padding: 0;
}

.c1 {
  max-width: 100%;
  margin-left: auto;
  margin-right: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c2 {
  width: 100%;
  margin-bottom: 0;
}

.c3 {
  padding: 16px;
}

.c4 {
  margin-left: 0;
  margin-right: 0;
  display: block;
  height: 1px;
  background-color: var(--borderColor-default,var(--color-border-default,#d0d7de));
  margin-top: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 100%;
  -ms-flex: 1 1 100%;
  flex: 1 1 100%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  max-width: 100%;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  min-width: 1px;
}

.c7 {
  width: 100%;
  max-width: 1012px;
  margin-left: auto;
  margin-right: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 16px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  width: 100%;
  margin-left: 0;
  margin-right: 0;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  margin-bottom: 0;
}

.c9 {
  margin-left: 0;
  margin-right: 0;
  display: none;
  margin-top: 0;
}

.c10 {
  --pane-min-width: 256px;
  --pane-max-width-diff: 511px;
  --pane-max-width: calc(100vw - var(--pane-max-width-diff));
  width: 100%;
  padding: 16px;
}

.c11 {
  height: 100%;
  position: relative;
  display: none;
  margin-left: 0;
}

.c12 {
  -webkit-order: 4;
  -ms-flex-order: 4;
  order: 4;
  width: 100%;
  margin-top: 0;
}

.c13 {
  margin-left: 0;
  margin-right: 0;
  display: block;
  height: 1px;
  background-color: var(--borderColor-default,var(--color-border-default,#d0d7de));
  margin-bottom: 0;
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {

}

@media screen and (min-width:1012px) {
  .c3 {
    padding: 24px;
  }
}

@media screen and (min-width:768px) {
  .c4 {
    margin-left: 0 !important;
    margin-right: 0 !important;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {

}

@media screen and (min-width:1012px) {
  .c7 {
    padding: 24px;
  }
}

@media screen and (min-width:768px) {
  .c8 {
    width: auto;
    margin-top: 0 !important;
    margin-bottom: 0 !important;
    position: -webkit-sticky;
    position: sticky;
    top: 0px;
    max-height: var(--sticky-pane-height);
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    margin-right: 0;
  }
}

@media screen and (min-width:768px) {
  .c9 {
    margin-left: 0 !important;
    margin-right: 0 !important;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {
  .c10 {
    width: 256px;
    overflow: auto;
  }
}

@media screen and (min-width:1012px) {
  .c10 {
    width: 296px;
    padding: 24px;
  }
}

@media screen and (min-width:1280px) {
  .c10 {
    --pane-max-width-diff: 959px;
  }
}

@media screen and (min-width:768px) {
  .c13 {
    margin-left: 0 !important;
    margin-right: 0 !important;
  }
}

<div>
  <div
    class="c0"
    style="--sticky-pane-height: 100vh;"
  >
    <div
      class="c1"
    >
      <header
        class="c2"
      >
        <div
          class="c3"
        >
          Header
        </div>
        <div
          class="c4"
        />
      </header>
      <div
        class="c5"
      >
        <main
          class="c6"
        >
          <div
            class=""
          />
          <div
            class="c7"
          >
            Content
          </div>
          <div
            class=""
          />
        </main>
        <div
          class="c8"
        >
          <div
            class="c9"
          />
          <div
            class="c10"
            style="--pane-width: 296px;"
          >
            Pane
          </div>
          <div
            class="c11"
          />
        </div>
      </div>
      <footer
        class="c12"
      >
        <div
          class="c13"
        />
        <div
          class="c3"
        >
          Footer
        </div>
      </footer>
    </div>
  </div>
</div>
`;
